### PR TITLE
ENH: Fix a 3D camera within FixedReference frame.

### PR DESCRIPTION
**Pull request by MichaelColonel**
The corresponding CheckBox widget is added to the "Parameter set" section of the module GUI.

Possible solution for #246.

### CODE REVIEW

**Comment by cpinter** [comment]
Thank you! Please include a video showing the change as you intend it to facilitate the review process.

**Comment by MichaelColonel** [comment]
Small demo: https://rutube.ru/video/private/1dc1159c438bd62fcb329fad31e1f5b0/?r=wd&p=W7L6xEmfVMXLodEleyoFJw

**Comment by cpinter** [comment]
Thanks a lot! Do you know why that initial jump happens? (at 19s)

**Comment by MichaelColonel** [comment]
I don't know the reason.

I've noticed that bug, for some reason it jumps only first time when transform node is updated, and it doesn't jump any longer, even if the check box is unchecked and checked once again.

**Comment by lassoan** [comment]
It looks very nice!

I've also noticed some transient jumping around while the slider is moved. You can prevent that by calling `pauseRender`, update all the transforms, then call `resumeRender`.

**Comment by MichaelColonel** [comment]
> You can prevent that by calling `pauseRender`, update all the transforms, then call `resumeRender`.

I will try to test that.

**Comment by MichaelColonel** [comment]
Most problems are fixed.
Small Demo2 : https://rutube.ru/video/private/7e73c212d9cc3e740a74bf8e0a3ad585/?r=wd&p=UkxnpbJlpIfw4qA_28JQoA

**Comment by MichaelColonel** [comment]
Ready to review and testing.

File: RoomsEyeView/qSlicerRoomsEyeViewModuleWidget.cxx [hunk 1]
```diff
@@ -152,6 +154,47 @@ vtkMRMLRTPlanNode* qSlicerRoomsEyeViewModuleWidgetPrivate::currentPlanNode(vtkMR
...
 
+//-----------------------------------------------------------------------------
+vtkMRMLCameraNode* qSlicerRoomsEyeViewModuleWidgetPrivate::get3DViewCameraNode() const
+{
+  Q_Q(const qSlicerRoomsEyeViewModuleWidget);
+
+  // Get 3D view node
+  qSlicerApplication* slicerApplication = qSlicerApplication::application();
+  qSlicerLayoutManager* layoutManager = slicerApplication->layoutManager();
+  qMRMLThreeDView* threeDView = layoutManager->threeDWidget(0)->threeDView();
```
Comment (cpinter) [inline RoomsEyeView/qSlicerRoomsEyeViewModuleWidget.cxx:170]: Please check threeViewCount and if it is 0 return nullptr here.

File: RoomsEyeView/qSlicerRoomsEyeViewModuleWidget.cxx [hunk 2]
```diff
@@ -996,3 +1035,28 @@ void qSlicerRoomsEyeViewModuleWidget::updateTreatmentOrientationMarker()
...
+
+  // Get 3D view node camera
+  vtkMRMLCameraNode* cameraNode = d->get3DViewCameraNode();
+  if (!cameraNode)
+  {
+    return;
+  }
+
+  vtkMRMLLinearTransformNode* node = d->logic()->GetTransformNodeBetween(vtkSlicerIECTransformLogic::FixedReference, vtkSlicerIECTransformLogic::RAS);
+  vtkNew<vtkMatrix4x4> transformMatrix;
```
Comment (cpinter) [inline RoomsEyeView/qSlicerRoomsEyeViewModuleWidget.cxx]: Give this transform a meaningful name please

**Review by cpinter — CHANGES_REQUESTED** [review]
Sorry for the delay! I tested the branch and it works great! I just added two very small comments about the code. Thank you very much!

**Comment by MichaelColonel** [comment]
Fixes were added.

**Comment by cpinter** [comment]
Perfect, thank you so much!

**Comment by ferdymercury** [comment]
Thanks a lot for the implementation Michael and Csaba/lassoan for the review.

**Pull request merged**